transport/tchannel/options.go

```diff
@@ -107,3 +112,11 @@ func ServiceName(name string) TransportOption {
 		t.name = name
 	}
 }
+
+// ConnectionTimeout specifies the time that TChannel will wait for a
+// connection attempt to any retained peer.
+func ConnectionTimeout(d time.Duration) TransportOption {
+	return func(t *transportConfig) {
```

Comment:
you use struct pointer here, i think `newTransportConfig` should return one as well